### ISSUE TITLE
Group constraints fail with inexistent location

### ISSUE THREAD

**Issue by timtroendle** [calliope-project/calliope#198]
## Problem description

When a group constraint contains a location that does not exist, Pyomo raises an error and Calliope crashes. Apart from the fact that there should probably be a helpful error message, I wonder: is the crash a bug or expected behaviour? 

Compare this to links: here, when one or both of the locations do not exist, Calliope issues a warning and ignores the link.

## Steps to reproduce the problem

Have a group constraint like this one:

```yaml
group_constraints:
    renewable_share_alps:
        locs: ["AUT", "CHE"]
        techs: ["wind", "pv"]
        demand_share_min:
            electricity: 0.20
```

With the location "AUT" not in your model (in my case: removed in one scenario).

## Calliope version

[0.6.4-dev 8e08dc1](https://github.com/calliope-project/calliope/tree/8e08dc1e5c20173d6a8ba895714ab180cee26abf)

**Comment by brynpickering** [comment]
I would say that this shouldn't fail, but should raise a warning, in case a user simply misspelled the location. But, otherwise ignoring an inexistent location (and tech) is fine by me.

**Comment by sjpfenninger** [comment]
Fixed in #232
